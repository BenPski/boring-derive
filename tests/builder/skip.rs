use boring_derive::Builder;

#[derive(Debug, Default, Builder)]
struct Example {
    #[builder(skip)]
    field1: f32,
    field2: String,
}

fn main() {
    let example = Example::default().field1(0.0).field2("something");
    println!("{:?}", example);
}
