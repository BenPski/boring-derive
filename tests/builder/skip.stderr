error[E0599]: no method named `field1` found for struct `Example` in the current scope
  --> tests/builder/skip.rs:11:38
   |
4  | struct Example {
   | -------------- method `field1` not found for this struct
...
11 |     let example = Example::default().field1(0.0).field2("something");
   |                                      ^^^^^^ field, not a method
   |
help: remove the arguments
   |
11 -     let example = Example::default().field1(0.0).field2("something");
11 +     let example = Example::default().field1.field2("something");
   |
help: there is a method `field2` with a similar name
   |
11 |     let example = Example::default().field2(0.0).field2("something");
   |                                      ~~~~~~
